### ISSUE TITLE
Podman Windows v4.2.1 msi not able create api forwarding

### ISSUE THREAD

**Issue by waylew-abc** [containers/podman#16136]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

**Steps to reproduce the issue:**

1. install podman msi

2. run 'podman machine init'

3.run 'podman machine start'

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
podman.exe version 4.2.1
```

**Output of `podman info`:**

```
host:
  arch: amd64
  buildahVersion: 1.27.0
  cgroupControllers:
  - cpuset
  - cpu
  - cpuacct
  - blkio
  - memory
  - devices
  - freezer
  - net_cls
  - perf_event
  - net_prio
  - hugetlb
  - pids
  - rdma
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.4-2.fc36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.4, commit: '
  cpuUtilization:
    idlePercent: 99.18
    systemPercent: 0.41
    userPercent: 0.41
  cpus: 4
  distribution:
    distribution: fedora
    variant: container
    version: "36"
  eventLogger: journald
  hostname: PF3GR8W5
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.10.102.1-microsoft-standard-WSL2
  linkmode: dynamic
  logDriver: journald
  memFree: 7906738176
  memTotal: 10437439488
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.6-2.fc36.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.6
      commit: 18cf2efbb8feb2b2f20e316520e0fd0b6c41ef4d
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.x86_64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 3220164608
  swapTotal: 3221225472
  uptime: 2h 37m 18.00s (Approximately 0.08 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /usr/share/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphRootAllocated: 269490393088
  graphRootUsed: 374198272
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 4.2.1
  Built: 1662580699
  BuiltTime: Wed Sep  7 15:58:19 2022
  GitCommit: ""
  GoVersion: go1.18.5
  Os: linux
  OsArch: linux/amd64
  Version: 4.2.1
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman` or `brew info podman`):**

```
(paste your output here)
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by waylew-abc** [comment]
interesting, now I get this error when running 'podman machine start' $ podman machine start
Starting machine "podman-machine-default"
API forwarding listening on: npipe:////./pipe/podman-machine-default

Another process was listening on the default Docker API pipe address.
You can still connect Docker API clients by setting DOCKER HOST using the
following powershell command in your terminal session:

        $Env:DOCKER_HOST = 'npipe:////./pipe/podman-machine-default'

Or in a classic CMD prompt:

        set DOCKER_HOST = 'npipe:////./pipe/podman-machine-default'

**Comment by waylew-abc** [comment]
hmm... unable to pull image with podman, but can with docker cli  "podman run -dt -p 8080:80/tcp docker.io/library/httpd
Trying to pull docker.io/library/httpd:latest...
Error: initializing source docker://httpd:latest: pinging container registry registry-1.docker.io: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io: Temporary failure in name resolution"

**Comment by rhatdan** [comment]
@waylew-abc @n1hility Is this still an issue?

**Comment by TheRealHnefi** [comment]
@rhatdan I have the same issue. Running podman 4.3.0. It seems to randomly cycle between this error and the following output:

API forwarding for Docker API clients is not available due to the following startup failures.
        CreateFile \\.\pipe\docker_engine: All pipe instances are busy.

I'm guessing some other application or service is using that pipe, but I don't know what application that would be.

**Comment by n1hility** [comment]
The original issue in this report looks like #14811 , which should be fixed in 4.3

The issue that @waylew-abc is describing is unrelated, and looks like a local networking/dns issue 

@TheRealHnefi do you have Docker Desktop or Rancher or something else that might bind docker_engine running?

**Comment by TheRealHnefi** [comment]
@n1hility I have Docker Engine (without the Desktop option) installed, but the daemon is not running. Is there a way to determine which application is doing the binding, or perhaps force all applications to relinquish that resource? Or, if all else fails, make podman use another pipe?

**Comment by n1hility** [comment]
@TheRealHnefi sysinternals process explorer will tell you in most cases. You can search for open handle for "docker_engine" . However, it's also possible that the failure is a temporary false alarm and the actual binding was successful. You can try to do a "docker info" for example and if it returns successfully with podman info in the output it is actually working. 

One thing to check would be the proxy process is binding ok on the listening side

```
Get-WinEvent -ProviderName '.NET Runtime' -Oldest
```

The pipe busy error can also just mean "not ready", and there isn't a way to tell the difference. Currently podman will wait up to 20 seconds for it to become available (as long as you are on 4.3 or later), before giving up and reporting the last error.

Let me know what you see.

**Comment by n1hility** [comment]
Closing assuming this is no longer an issue. Feel free to reopen or create a new issue if there are more details to share.
